src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -54,17 +62,33 @@ public partial struct SyntaxValueProvider
         // changed. CreateSyntaxProvider will have to rerun all incremental nodes since it passes along the
         // SemanticModel, and that model is updated whenever any tree changes (since it is tied to the compilation).
         var syntaxTreesProvider = _context.CompilationProvider
-            .SelectMany(static (c, _) => c.SyntaxTrees)
+            .SelectMany((compilation, cancellationToken) =>
+            {
+                var count = 0;
+                foreach (var tree in compilation.CommonSyntaxTrees)
+                {
+                    var info = tree.GetInfo(syntaxHelper, cancellationToken);
+                    if (info.ContainsGlobalAliases || info.ContainsAttributeList)
+                        count++;
+                }
+
+                var builder = ImmutableArray.CreateBuilder<SyntaxTreeInfo>(count);
+
+                foreach (var tree in compilation.CommonSyntaxTrees)
+                {
+                    var info = tree.GetInfo(syntaxHelper, cancellationToken);
+                    if (info.ContainsGlobalAliases || info.ContainsAttributeList)
+                        builder.Add(info);
+                }
+
+                return builder.MoveToImmutable();
```

Comment:
expanded otu the linq version so we can preallocate the right sized arrays without garbage.  Also, because the tree info is on the tree itself, we avoid two full rounds of hashing + using the CWT from before.